source/fundamentals/data-formats/serialization.txt

```diff
@@ -0,0 +1,190 @@
+.. _fundamentals-kotlin-serialization:
+
+====================
+Kotlin Serialization
+====================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+The Kotlin driver supports the ``kotlinx.serialization`` library for serializing 
+and deserializing Kotlin objects. 
+
+The driver provides an efficient ``Bson`` serializer that can can be used with 
+``@Serializable`` classes to handle the serialization of Kotlin objects to BSON data. 
+The ``bson-kotlinx`` library also supports :ref:`custom codecs <kotlin-custom-codec>`
+with configurations such as encoding defaults, encoding nulls, and defining class 
+discriminators.
+
+.. note::
+
+    To use the ``Codec`` interface to specify custom encoding and decoding of 
+    Kotlin objects to BSON data, see the :ref:`Codecs <fundamentals-codecs>` 
+    documentation.
+
+Although you can use the Kotlin driver with the Kotlin Serialization's ``Json`` 
+library, the ``Json`` serializer does *not* directly support BSON value types such
+as ``ObjectId``. You must provide a custom serializer that can handle the 
+conversion between BSON and JSON. 
+
+Supported Types
+~~~~~~~~~~~~~~~
+
+The Kotlin driver supports all Kotlin types that are supported by the Kotlin 
+Serialization library and the following BSON types:
+
+- ObjectId
+- BsonNull
+- BsonArray
+- BsonBinary
+- BsonBoolean
+- BsonDateTime
+- BsonDbPointer
+- BsonDocument
+- BsonDouble
+- BsonInt32
+- BsonInt64
+- BsonDecimal128
+- BsonMaxKey
+- BsonMinKey
+- BsonJavaScript
+- BsonJavaScriptWithScope
+- BsonObjectId
+- BsonRegularExpression
+- BsonString
+- BsonSymbol
+- BsonTimestamp
+- BsonUndefined
+- BsonDocument
+- RawBsonDocument
+- RawBsonArray
+
+Any additional types can be supported using custom serializers. See the 
+`official Kotlin Documentation <https://kotlinlang.org/docs/serialization.html>`__ 
+for more information.
+
+Add Kotlin Serialization to Your Project
+----------------------------------------
+
+The Kotlin driver serialization support depends on the the official Kotlin 
+Serialization library. You must add `Kotlin Serialization <https://github.com/Kotlin/kotlinx.serialization>`__ 
```

Comment:
S: I think we should refer to this library as either Kotlin serialization library (lowercase s) or as the `kotlinx.serialization` library

Reply:
updated